extensions/interactions/GraphInput/directives/oppia-interactive-graph-input.component.ts

```diff
@@ -0,0 +1,185 @@
+  interactionIsActive: boolean = false;
+  errorMessage: string = '';
+
+  constructor(
+    private currentInteractionService: CurrentInteractionService,
+    private graphInputRulesService: GraphInputRulesService,
+    private interactionAttributesExtractorService:
+      InteractionAttributesExtractorService,
+    private playerPositionService: PlayerPositionService
+  ) {}
+
+  ngOnInit(): void {
+    this.componentSubscriptions.add(
+      this.playerPositionService.onNewCardAvailable.subscribe(
+        () => {
+          this.interactionIsActive = false;
+
+          this.canAddVertex = false;
+          this.canDeleteVertex = false;
+          this.canEditVertexLabel = false;
+          this.canMoveVertex = false;
+          this.canAddEdge = false;
+          this.canDeleteEdge = false;
+          this.canEditEdgeWeight = false;
+        }
+      )
+    );
+
+    this.errorMessage = '';
+    this.graph = {
+      vertices: [],
+      edges: [],
+      isDirected: false,
+      isWeighted: false,
+      isLabeled: false
+    };
+    this.interactionIsActive = (this.lastAnswer === null);
+
+    this.currentInteractionService.registerCurrentInteraction(
+      () => this.submitGraph(), () => this.validityCheckFn());
+
+    if (this.interactionIsActive) {
+      this.resetGraph();
+    } else {
+      this.graph = this.lastAnswer;
+    }
+    const {
+      canAddVertex,
+      canDeleteVertex,
+      canEditVertexLabel,
+      canMoveVertex,
+      canAddEdge,
+      canDeleteEdge,
+      canEditEdgeWeight
+    } = this.interactionAttributesExtractorService.getValuesFromAttributes(
+      'GraphInput',
+      this._getAttrs()
+    ) as GraphInputCustomizationArgs;
+    this.canAddVertex = this.interactionIsActive ? canAddVertex.value : false;
+    this.canDeleteVertex = this.interactionIsActive ?
+      canDeleteVertex.value : false;
+    this.canEditVertexLabel = this.interactionIsActive ?
+      canEditVertexLabel.value : false;
+    this.canMoveVertex = this.interactionIsActive ?
+      canMoveVertex.value : false;
+    this.canAddEdge = this.interactionIsActive ?
+      canAddEdge.value : false;
+    this.canDeleteEdge = this.interactionIsActive ?
+      canDeleteEdge.value : false;
+    this.canEditEdgeWeight = this.interactionIsActive ?
+      canEditEdgeWeight.value : false;
+  }
+
+  private _getAttrs() {
+    return {
+      graphWithValue: this.graphWithValue,
+      canAddVertexWithValue: this.canAddVertexWithValue,
+      canDeleteVertexWithValue: this.canDeleteVertexWithValue,
+      canMoveVertexWithValue: this.canMoveVertexWithValue,
+      canEditVertexLabelWithValue: this.canEditVertexLabelWithValue,
+      canAddEdgeWithValue: this.canAddEdgeWithValue,
+      canDeleteEdgeWithValue: this.canDeleteEdgeWithValue,
+      canEditEdgeWeightWithValue: this.canEditEdgeWeightWithValue
+    };
+  }
+
+  submitGraph(): void {
+    // Here, angular.copy is needed to strip $$hashkey from the graph.
```

Comment:
This comment is no longer entirely true.

Reply:
Changed it